src/Smalot/PdfParser/Header.php

```diff
@@ -60,6 +60,13 @@ public function __construct($elements = [], Document $document = null)
         $this->document = $document;
     }
 
+    public function init()
+    {
+        foreach ($this->elements as $name => $element) {
+            $element->init();
+        }
```

Comment:
Can we do that in the constructor instead?

Reply:
As far as I see, the elements are never set by anything other than the constructor. That could be an argument for moving it to the constructor, right after the elements are set.

However, this could hypothetically have a performance impact. Right now, this init method is explicitly called from the `Document`'s method of the same name, which is called from its method `setObjects()`. Imagine a scenario where you bulk-process hundrets or thousands of PDF files, where many of them will expectedly fail (secured pdfs, other parsing errors, anything that can happen before `setObjects()` is called within `parseContent()` of the `Parser`...). The `init` of every `Encoding` and `Font` that is encountered will be called as soon as the header is built. And that looks quite expensive, with several RegExes used especially for `Font`s, where `loadTranslateTable()` is called from within init.

That case might be quite constructed and improbable, but I wouldn't want to risk it just for saving a few lines of code.
Also, I think when Sebastien implemented it that way for `$object->init()` in `Document->init()`, which could probably also be moved to the object's constructor, he did that on purpose, so I'd tend to keep it that way for the header as well!

Reply:
Good arguments, so we keep it as it is.